src/software/jetson_nano/thunderloop.cpp

```diff
@@ -180,12 +182,10 @@ Thunderloop::~Thunderloop() {}
 
                 // Handle emergency stop override
                 auto nanoseconds_elapsed_since_last_primitive =
-                    time_since_last_primitive_received.tv_sec *
-                        static_cast<int>(NANOSECONDS_PER_SECOND) +
-                    time_since_last_primitive_received.tv_nsec;
+                    getMilliseconds(time_since_last_primitive_received);
 
                 if (nanoseconds_elapsed_since_last_primitive >
-                    static_cast<long>(PRIMITIVE_MANAGER_TIMEOUT_NS))
+                    PRIMITIVE_MANAGER_TIMEOUT_NS)
```

Comment:
The value is being assigned time in ms, but its name has nanoseconds. We should probably update the variable and potentially the timeout constant to also use ms.

Reply:
sorry the function change messed some stuff up, I added 2 functions for milliseconds and nanoseconds each since we need both values in different places